### PR TITLE
Disallow duplicate lessons

### PR DESCRIPTION
**Pull request by ptejasv**
# Changelog
- Print a message "There are no modules in the list." when the module list is empty
- Disallow the user to add a duplicate lesson with the same title on the same day of the week
- Add tests to check that duplicate lessons cannot be added

Duplicate module addition has already been handled.

### CODE REVIEW

File: src/main/java/seedu/duke/model/lesson/LessonList.java [hunk 1]
```diff
@@ -38,7 +39,13 @@ public boolean isEmpty() {
         return lessonList.isEmpty();
     }
 
-    public void addLesson(Lesson newLesson) {
+    public void addLesson(Lesson newLesson) throws DukeException {
+        for (Lesson lesson : lessonList) {
+            if (lesson.getTitle().equals(newLesson.getTitle())
+                    && (lesson.getDayOfTheWeek().equals(newLesson.getDayOfTheWeek()))) {
```
Comment (richwill28) [inline src/main/java/seedu/duke/model/lesson/LessonList.java:45]: Should probably check if the start and end time are similar as well? Also, consider refactoring the logic inside the `if` for SLAP.

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java [hunk 2]
```diff
@@ -21,12 +21,16 @@ public ListModuleCommand(String argument) {
...
-        assert (argument.equalsIgnoreCase("verbose") | argument.isEmpty())
-                : "list module command currently only accepts empty, \"verbose\"";
-        if (argument.equalsIgnoreCase("verbose")) {
-            ui.printModulesWithDetails(moduleList);
+        if (moduleList.isEmpty()) {
+            ui.printMessage("There are no modules in the list.");
         } else {
-            ui.printModuleList(moduleList);
+            assert (argument.equalsIgnoreCase("verbose") | argument.isEmpty())
+                    : "list module command currently only accepts empty, \"verbose\"";
```
Comment (richwill28) [inline src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:28]: Faulty input should be handled with exception instead?

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java [hunk 3]
```diff
@@ -21,12 +21,16 @@ public ListModuleCommand(String argument) {
     @Override
     public void execute(Ui ui, Storage storage, TaskList taskList, LessonList lessonList, ModuleList moduleList)
             throws DukeException, IOException {
-        assert (argument.equalsIgnoreCase("verbose") | argument.isEmpty())
-                : "list module command currently only accepts empty, \"verbose\"";
-        if (argument.equalsIgnoreCase("verbose")) {
-            ui.printModulesWithDetails(moduleList);
+        if (moduleList.isEmpty()) {
+            ui.printMessage("There are no modules in the list.");
```
Comment (richwill28) [inline src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:25]: I think this should be handled by an exception.

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java [hunk 4]
```diff
@@ -21,12 +21,16 @@ public ListModuleCommand(String argument) {
...
         } else {
-            ui.printModuleList(moduleList);
+            assert (argument.equalsIgnoreCase("verbose") | argument.isEmpty())
+                    : "list module command currently only accepts empty, \"verbose\"";
+            if (argument.equalsIgnoreCase("verbose")) {
+                ui.printModulesWithDetails(moduleList);
+            } else {
+                ui.printModuleList(moduleList);
+            }
```
Comment (richwill28) [inline src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:33]: Hmm.. any good reason of nesting this inside an `else` rather than just using `else if`?

File: src/test/java/seedu/duke/model/lesson/LessonTest.java [hunk 5]
```diff
@@ -4,14 +4,19 @@
...
-        lessonList.addLesson(new Lesson("CS1101S Lecture", "Wednesday", "10:00 PM", "12:00 PM", "-"));
+        try {
+            lessonList.addLesson(new Lesson("CS3243 Lecture", "Monday", "10:00 PM", "12:00 PM", " "));
+            lessonList.addLesson(new Lesson("CS1101S Lecture", "Wednesday", "10:00 PM", "12:00 PM", "-"));
+        } catch (DukeException e) {
+            fail();
+        }
 
         assertThrows(DukeException.class, () -> lessonList.getLesson(1).launchUrl());
         assertThrows(DukeException.class, () -> lessonList.getLesson(2).launchUrl());
```
Comment (richwill28) [inline src/test/java/seedu/duke/model/lesson/LessonTest.java:22]: Is it possible to enclose everything here with `try-catch`? Just to give more emphasis on the happy path.

File: src/test/java/seedu/duke/model/lesson/LessonListTest.java [hunk 6]
```diff
@@ -19,7 +19,11 @@ public void isEmpty_emptyList_true() {
     @Test
     public void isEmpty_nonEmptyList_false() {
         LessonList lessonList = new LessonList();
-        lessonList.addLesson(new Lesson("CS2113T Lecture", "Monday", "02:00 PM", "04:00 PM", "-"));
+        try {
+            lessonList.addLesson(new Lesson("CS2113T Lecture", "Monday", "02:00 PM", "04:00 PM", "-"));
+        } catch (DukeException e) {
+            fail();
+        }
         assertFalse(lessonList.isEmpty());
```
Comment (richwill28) [inline src/test/java/seedu/duke/model/lesson/LessonListTest.java:27]: Same here.

File: src/test/java/seedu/duke/logic/commands/lesson/AddLessonCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+
+import org.junit.jupiter.api.Test;
+import seedu.duke.DukeException;
+import seedu.duke.model.lesson.Lesson;
+import seedu.duke.model.lesson.LessonList;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
```
Comment (richwill28) [inline src/test/java/seedu/duke/logic/commands/lesson/AddLessonCommandTest.java:11]: Remember to add your authorship :)

**Review by richwill28 — APPROVED** [review]
LGTM, just a few nits.

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java (hunk 3 above)
Comment (ptejasv) [reply src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:25]: I'm not sure about this. ListTaskCommand and ListLessonCommand both handle it using an if-check and I don't think the list being empty isn't exceptional behaviour, it's normal for the list to be empty?

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java (hunk 4 above)
Comment (ptejasv) [reply src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:33]: This actually wasn't written by me but I see your point, I'll change this.

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java (hunk 2 above)
Comment (ptejasv) [reply src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:28]: This is also not written by me but yep I can change it.

File: src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java (hunk 3 above)
Comment (richwill28) [reply src/main/java/seedu/duke/logic/commands/module/ListModuleCommand.java:25]: > the list being empty isn't exceptional behaviour

Good point. The current implementation is fine then.
